js/modules/Debug/Debug.js

```diff
@@ -0,0 +1,1380 @@
+                     <li class="nav-item"><a class="nav-link active" data-bs-toggle="tab" href="#debugpost${rand}">POST</a></li>
+                     <li class="nav-item"><a class="nav-link" data-bs-toggle="tab" href="#debugget${rand}">GET</a></li>
+                     <li class="nav-item"><a class="nav-link" data-bs-toggle="tab" href="#debugsession${rand}">SESSION</a></li>
+                     <li class="nav-item"><a class="nav-link" data-bs-toggle="tab" href="#debugserver${rand}">SERVER</a></li>
+                  </ul>
+            
+                  <div class="card-body overflow-auto p-1">
+                     <div class="tab-content">
+                        <div id="debugpost${rand}" class="cm-s-default tab-pane active"></div>
+                        <div id="debugget${rand}" class="cm-s-default tab-pane"></div>
+                        <div id="debugsession${rand}" class="cm-s-default tab-pane"></div>
+                        <div id="debugserver${rand}" class="cm-s-default tab-pane"></div>
+                     </div>
+                  </div>
+               </div>
+            </div>
+        `);
+
+        const selected_request_id = content_area.data('request_id');
+
+        const matching_profile = this.getProfile(selected_request_id);
+        const globals = matching_profile.globals;
+        appendGlobals(globals['post'], content_area.find(`#debugpost${rand}`));
+        appendGlobals(globals['get'], content_area.find(`#debugget${rand}`));
+        appendGlobals(globals['session'], content_area.find(`#debugsession${rand}`));
+        appendGlobals(globals['server'], content_area.find(`#debugserver${rand}`));
+
+        content_area.on('shown.bs.tab', 'a[data-bs-toggle="tab"]', (e) => {
+            const target = $(e.target).attr('href');
+            const target_el = content_area.find(target);
+            const previously_shown = target_el.data('previously_shown') || false;
+            const editor = target_el.data('editor');
+            if (!previously_shown && editor) {
+                editor.refresh();
+
+                setTimeout(() => {
+                    // Stupid solution to fold all levels except the first one.
+                    // foldCode(0), would fold the first level only and doesn't handle nested levels.
+                    const total_lines = editor.lineCount();
+                    // Must start from the bottom, otherwise it doesn't fold parent levels
+                    for (let i = total_lines - 1; i > 1; i--) {
+                        editor.foldCode(window.CodeMirror.Pos(i, 0));
+                    }
+                }, 100);
+            }
+            target_el.data('previously_shown', true);
+        });
+
+        // trigger shown.bs.tab on the first tab manually since the event is not triggered on page load
+        content_area.find('a[data-bs-toggle="tab"]').first().trigger('shown.bs.tab');
+    }
+
+    showClientPerformance(content_area, refresh = false) {
```

Comment:
@trasher said the same thing this morning.
I passed initialy as i was not interrested by the content.
But I agree, we may remove if it's duplicate information.

@cconard96 do you have some explanations about the purpose ?

Reply:
It was added for the same reason the `dev` plugin added a tab to the original debug panel for browser console output.
Some people choose to use GLPI from phones and tablets and the mobile version of browsers don't have developer tools. If an issue exists in those types of browsers, it can be difficult to debug.

Reply:
ok seems legit so.
Could you at least hide the tab when the browser can't display anything (firefox for example)

Reply:
Firefox can. For some reason, it doesn't record a "first-paint" timing and it was throwing an error. I added a fix/fallback for it.